ui/src/app/views/concept-homepage/component.ts

```diff
@@ -0,0 +1,152 @@
+  conceptAddModal: ConceptAddModalComponent;
+
+  conceptDomainList: Array<DomainInfo> = [];
+
+  concepts: Array<Concept> = [];
+  conceptsCache: Array<ConceptCacheSet> = [];
+
+  wsNamespace: string;
+  wsId: string;
+
+  constructor(
+    private conceptsService: ConceptsService,
+    private route: ActivatedRoute,
+  ) {
+    this.wsNamespace = this.route.snapshot.params['ns'];
+    this.wsId = this.route.snapshot.params['wsid'];
+  }
+
+  ngOnInit(): void {
+    this.loadingDomains = true;
+    this.conceptsService.getDomainInfo(this.wsNamespace, this.wsId).subscribe((response) => {
+      this.conceptDomainList = response.items;
+      this.conceptDomainList.forEach((domain) => {
+        this.conceptsCache.push({
+          domain: domain.domain,
+          items: []
+        });
+      });
+      this.loadingDomains = false;
+      this.selectedDomain = this.conceptDomainList[0].domain;
+    });
+  }
+
+  openAddModal(): void {
+    this.conceptAddModal.open();
+  }
+
+  selectDomain(domain: Domain) {
+    this.selectedDomain = domain;
+    this.setConcepts();
+  }
+
+  searchButton() {
+    this.currentSearchString = this.searchTerm;
+    this.searchConcepts();
+  }
+
+  browseDomain(domain: DomainInfo) {
+    this.currentSearchString = '';
+    this.selectedDomain = domain.domain;
+    this.searchConcepts();
+  }
+
+  searchConcepts() {
+    console.log(this.standardConceptsOnly);
+    this.searching = true;
+    this.searchLoading = true;
+    let standardConceptFilter: StandardConceptFilter;
+    if (this.standardConceptsOnly) {
+      standardConceptFilter = StandardConceptFilter.STANDARDCONCEPTS;
+    } else {
+      standardConceptFilter = StandardConceptFilter.ALLCONCEPTS;
+    }
+
+    let numCalls = 0;
+    const numCallsSubject = new BehaviorSubject<number>(0);
+    const numCallsSubject$ = numCallsSubject.asObservable();
+
+    numCallsSubject$.subscribe((finishedCalls) => {
+      if (finishedCalls === this.conceptsCache.length) {
+        this.searchLoading = false;
+        this.setConcepts();
+      }
+    });
+
+    this.conceptsCache.forEach((conceptDomain) => {
+      const request = {
+        query: this.currentSearchString,
+        standardConceptFilter: standardConceptFilter,
+        domain: conceptDomain.domain
+      };
+      this.conceptsService.searchConcepts(this.wsNamespace, this.wsId, request)
+        .subscribe((response) => {
+          numCalls += 1;
+          numCallsSubject.next(numCalls);
+          conceptDomain.items = response.items;
+      });
+    });
+  }
+
+  setConcepts() {
+    this.concepts = this.conceptsCache.find(
+      conceptDomain => conceptDomain.domain === this.selectedDomain).items;
+  }
+
+  get getSearchDisabled() {
+    return this.searchLoading || this.loadingDomains || this.searchTermNotLongEnough;
+  }
+
+  get searchTermNotLongEnough() {
+    return this.searchTerm === undefined || this.searchTerm.length < 3;
+  }
+
+  setStandardOnly() {
```

Comment:
Is this used? Stub out code?  I can see how you would use it to reload the search results from cache or launch a new search

Reply:
This is old from previous work. Can remove.